mm/alignment.py

```diff
@@ -391,6 +391,114 @@ def hash(cls, kwargs):
+        :param zoom_x_max: Maximum X value in which to zoom in on the plot. If `None`, won't zoom in.
+        :param zoom_y_min: Minimum Y value in which to zoom in on the plot. If `None`, won't zoom in.
+        :param zoom_y_max: Maximum Y value in which to zoom in on the plot. If `None`, won't zoom in.
+        :param num_bins: Number of histogram bins. By default `50`.
+        """
+        self.alignment_log_files = alignment_log_files
+        self.clip_low = clip_low
+        self.clip_high = clip_high
+        self.clip_percentile_low = clip_percentile_low
+        self.clip_percentile_high = clip_percentile_high
+        self.zoom_x_min = zoom_x_min
+        self.zoom_x_max = zoom_x_max
+        self.zoom_y_min = zoom_y_min
+        self.zoom_y_max = zoom_y_max
+        self.num_bins = num_bins
+
+        self.out_plot = self.output_path("plot.png")
+
+        self.rqmt = {"cpu": 1, "mem": 1.0, "time": 1.0}
+
+    def tasks(self):
+        yield Task("plot", resume="plot", rqmt=self.rqmt)
+
+    def plot(self):
+        import numpy as np
+        import matplotlib
+        import matplotlib.pyplot as plt
+
+        # Parse the files and search for the average alignment score values (normalized over time).
+        alignment_scores = []
+        for log_file in self.alignment_log_files:
+            logging.info("Reading: {}".format(log_file))
+            file_path = log_file.get_path()
+            document = ET.parse(util.uopen(file_path))
+            _seg_list = document.findall(".//segment")
+            for seg in _seg_list:
+                avg = seg.find(".//score/avg")
+                alignment_scores.append(float(avg.text))
+            del document
+
+        np_alignment_scores = np.asarray(alignment_scores)
+        min_value = np_alignment_scores.min()
+        max_value = np_alignment_scores.max()
+        logging.info("STATS:")
+        logging.info(f"Min: {min_value}")
+        logging.info(f"Median {np.median(np_alignment_scores)}")
+        logging.info(f"Max: {max_value}")
+        logging.info(f"Total number of segments: {np_alignment_scores.size}")
+
+        # Plot the data.
+        matplotlib.use("Agg")
+        if self.clip_low is not None and self.clip_percentile_low is not None:
+            # Plot the most restrictive (highest).
+            clip_value_low = np.percentile(self.clip_percentile_low or 0)
+            clip_low = self.clip_low if self.clip_low > clip_value_low else clip_value_low
+        else:
+            clip_low = self.clip_low or self.clip_percentile_low or min_value
+        if self.clip_high is not None and self.clip_percentile_high is not None:
+            # Plot the most restrictive (lowest).
+            clip_value_high = np.percentile(self.clip_percentile_high or 0)
+            clip_high = self.clip_high if self.clip_high < clip_value_high else clip_value_high
+        else:
+            clip_high = self.clip_high or self.clip_percentile_high or max_value
```

Comment:
```suggestion
        clip_low = max(self.clip_low, np.percentile(self.clip_percentile_low), min_value)
        clip_high = min(self.clip_high, np.percentile(self.clip_percentile_high), max_value)
```
the most restrictive shall prevail